### PR TITLE
Fix Minor memory leaks in test_buffer/compiler.

### PR DESCRIPTION
**Pull request by JohnKesapidesARM**
Fix vairous memory leaks in test_buffer and test_compiler.

Signed-off-by: John Kesapides <john.kesapides@arm.com>

### CODE REVIEW

File: test_conformance/compiler/test_compiler_defines_for_extensions.cpp [hunk 1]
```diff
@@ -335,10 +335,10 @@ int test_compiler_defines_for_extensions(cl_device_id device, cl_context context
     strcat(kernel_code, kernel_strings[4]);
 
     // Now we need to execute the kernel
-    cl_mem defines;
+    clMemWrapper defines;
```
Comment (jlewis-austin) [inline test_conformance/compiler/test_compiler_defines_for_extensions.cpp:338]: There's a redundant clReleaseMemObject at line 436 below.

**Review by jlewis-austin — CHANGES_REQUESTED** [review]
Thanks this looks good, just some minor nits.

File: test_conformance/buffers/test_buffer_fill.cpp [hunk 2]
```diff
@@ -619,16 +623,20 @@ int test_buffer_fill( cl_device_id deviceID, cl_context context, cl_command_queu
...
 
-            err = clEnqueueFillBuffer(queue, buffers[ii], pattern[i], ptrSizes[i],
-                                      ptrSizes[i] * offset_elements, ptrSizes[i] * fill_elements,
-                                      0, NULL, &(event[0]));
+            err = clEnqueueFillBuffer(
+                queue, buffers[0], pattern[i], ptrSizes[i],
+                ptrSizes[i] * offset_elements, ptrSizes[i] * fill_elements, 0,
+                NULL, &(event[0]));
             /* uncomment for test debugging
              err = clEnqueueWriteBuffer(queue, buffers[ii], CL_FALSE, 0, ptrSizes[i]*num_elements, inptr[i], 0, NULL, &(event[0]));
```
[the remark this answers is not in the document]
Comment (EwanC) [reply test_conformance/buffers/test_buffer_fill.cpp:641]: Also have a preference for removing it, and deleting the `ii` variable

File: test_conformance/buffers/test_buffer_write.cpp [hunk 3]
```diff
@@ -851,7 +883,9 @@ int test_buffer_write_struct( cl_device_id deviceID, cl_context context, cl_comm
 
                 if (gTestMap) {
                     void *dataPtr;
-                    dataPtr = clEnqueueMapBuffer(queue, buffers[ii], CL_TRUE, CL_MAP_WRITE, 0, ptrSizes[i]*num_elements, 0, NULL, NULL, &err);
+                    dataPtr = clEnqueueMapBuffer(
```
Comment (EwanC) [inline test_conformance/buffers/test_buffer_write.cpp:886]: Should the unmap on line 898 be updated to `buffers[0]` from `buffers[ii]` too?

Likewise the calls to `clSetKernelArg` use `buffers[ii]` on line 918+919(github won't let me comment directly on the lines)

**Comment by SebastianLuzynski** [comment]
This issue is also fixed by my PR #1162, that fixes flaw in wrapper implementation.

**Review by jlewis-austin — DISMISSED** [review]
LGTM

File: test_conformance/buffers/test_buffer_fill.cpp [hunk 4]
```diff
@@ -586,8 +585,6 @@ int test_buffer_fill( cl_device_id deviceID, cl_context context, cl_command_queu
     loops = (loops < 5 ? loops : 5);
     for (i = 0; i < loops; i++)
     {
-        ii = i << 1;
```
Comment (EwanC) [inline test_conformance/buffers/test_buffer_fill.cpp:589]: Think the `int ii` definition on line 571 can be deleted now too, LGTM otherwise

**Review by EwanC — APPROVED** [review]

**Review by jlewis-austin — APPROVED** [review]

**Pull request merged**